src/collect.jl

```diff
@@ -128,3 +128,41 @@ function widenarray(dest::AbstractArray, i, ::Type{T}) where T
     copyto!(new, 1, dest, 1, i-1)
     new
 end
+
+"""
+`collect_to_structarray!(dest, itr) -> dest′`
+
+Try to append `itr` into a vector `dest`.  Widen element type of
+`dest` if it cannot hold the elements of `itr`.  That is to say,
+
+```julia
+vcat(dest, StructVector(itr)) == collect_to_structarray!(dest, itr)
+```
+
+holds.  Note that `dest′` may or may not be the same object as `dest`.
+The state of `dest` is unpredictable after `collect_to_structarray!`
+is called (e.g., it may contain just half of the elements from `itr`).
+"""
+collect_to_structarray!(dest::AbstractVector, itr) =
```

Comment:
As mentioned in the discussion, maybe call this `append!!`, without exporting it, so that it's clearer what it does. `collect_to_structarray!` can be left as it was.